ffv1.md

```diff
@@ -0,0 +1,960 @@
+
+180,181,182,183,184,185,186,187,188,189,190,190,191,192,194,194,
+
+195,196,197,198,199,200,201,202,202,204,205,206,207,208,209,209,
+
+210,211,212,213,215,215,216,217,218,219,220,220,222,223,224,225,
+
+226,227,227,229,229,230,231,232,234,234,235,236,237,238,239,240,
+
+241,242,243,244,245,246,247,248,248,  0,  0,  0,  0,  0,  0,  0,
+```
+
+#### alternative state transition table
+
+The alternative state transition table has been build using iterative minimization of frame sizes and generally performs better than the default. To use it, the coder_type has to be set to 2 and the difference to the default has to be stored in the parameters. The reference implemenation of FFV1 in FFmpeg uses this table by default at the time of this writing when Range coding is used.
+
+```
+  0, 10, 10, 10, 10, 16, 16, 16, 28, 16, 16, 29, 42, 49, 20, 49,
+
+ 59, 25, 26, 26, 27, 31, 33, 33, 33, 34, 34, 37, 67, 38, 39, 39,
+
+ 40, 40, 41, 79, 43, 44, 45, 45, 48, 48, 64, 50, 51, 52, 88, 52,
+
+ 53, 74, 55, 57, 58, 58, 74, 60,101, 61, 62, 84, 66, 66, 68, 69,
+
+ 87, 82, 71, 97, 73, 73, 82, 75,111, 77, 94, 78, 87, 81, 83, 97,
+
+ 85, 83, 94, 86, 99, 89, 90, 99,111, 92, 93,134, 95, 98,105, 98,
+
+105,110,102,108,102,118,103,106,106,113,109,112,114,112,116,125,
+
+115,116,117,117,126,119,125,121,121,123,145,124,126,131,127,129,
+
+165,130,132,138,133,135,145,136,137,139,146,141,143,142,144,148,
+
+147,155,151,149,151,150,152,157,153,154,156,168,158,162,161,160,
+
+172,163,169,164,166,184,167,170,177,174,171,173,182,176,180,178,
+
+175,189,179,181,186,183,192,185,200,187,191,188,190,197,193,196,
+
+197,194,195,196,198,202,199,201,210,203,207,204,205,206,208,214,
+
+209,211,221,212,213,215,224,216,217,218,219,220,222,228,223,225,
+
+226,224,227,229,240,230,231,232,233,234,235,236,238,239,237,242,
+
+241,243,242,244,245,246,247,248,249,250,251,252,252,253,254,255,
+```
+
+### Huffman coding mode
+
+This coding mode uses golomb rice codes. The VLC code is split into 2 parts, the prefix stores the most significant bits, the suffix stores the k least significant bits or stores the whole number in the ESC case. The end of the bitstream (of the frame) is filled with 0-bits so that the bitstream contains a multiple of 8 bits.
+
+#### Prefix
+
+
+|bits           | value |
+|:--------------|:------| 
+|1              | 0     |
+|01             | 1     |
+|...            | ...   |
+|0000 0000 0001 | 11    |
+|0000 0000 0000 | ESC   |
+
+
+#### Suffix
+
+
+|              |                                                         |
+|:-------------|---------------------------------------------------------|
+|non ESC       | the k least significant bits MSB first|
+|ESC           | the value - 11, in MSB first order, ESC may only be used if the value cannot be coded as non ESC|
+
+#### Examples
+
+
+|k | bits | value|
+|:---:|:------------------------:|:-------:|
+|0 | 1 | 0|
+|0 | 001 | 2|
+|2 | 1 00 | 0|
+|2 | 1 10 | 2|
+|2 | 01 01 | 5|
+|any | 000000000000 10000000 | 139|
```

Comment:
Vertical alignment. Also I'd use <code>``</code> for the bits to make them line up.